### ISSUE TITLE
DIO请求本地https接口报错，postman测试正常，已排除证书问题

### ISSUE THREAD

**Issue by zlczhou** [flutterchina/dio#894]
### New Issue Checklist
2020-08-12 17:19:09 ERROR FEBS-Gateway boundedElastic-5 org.springframework.boot.autoconfigure.web.reactive.error.AbstractErrorWebExceptionHandler [5e9a238d-5]  500 Server Error for HTTP PRI ""
org.springframework.web.server.ResponseStatusException: 404 NOT_FOUND
	at org.springframework.web.reactive.resource.ResourceWebHandler.lambda$handle$0(ResourceWebHandler.java:325)
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Error has been observed at the following site(s):
	|_ checkpoint ⇢ org.springframework.cloud.gateway.filter.WeightCalculatorWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.security.web.server.authorization.AuthorizationWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.security.web.server.authorization.ExceptionTranslationWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.security.web.server.authentication.logout.LogoutWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.security.web.server.savedrequest.ServerRequestCacheWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.security.web.server.context.SecurityContextServerWebExchangeWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.security.web.server.context.ReactorContextWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.security.web.server.header.HttpHeaderWriterWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.security.config.web.server.ServerHttpSecurity$ServerWebExchangeReactorContextWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.security.web.server.WebFilterChainProxy [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.boot.actuate.metrics.web.reactive.server.MetricsWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ cc.mrbird.febs.gateway.enhance.auth.WebFluxSecurityCorsFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ HTTP null "" [ExceptionHandlingWebHandler]

- [ ] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |   1.20.1   |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |   3.0.10   |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |   4.0.1   |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps
dioutil代码：
class DioAppUtils {
  static DioAppUtils _instance;
  Dio _dio;
  BaseOptions _baseOptions;

  static DioAppUtils getInstance() {
    if (_instance == null) {
      _instance = new DioAppUtils();
    }
    return _instance;
  }

  /// dio初始化配置
  DioAppUtils() {
    //请求参数配置
    _baseOptions = new BaseOptions(
      baseUrl: "https://192.168.128.120:8443/app",
      connectTimeout: 5000,
      receiveTimeout: 3000,
      headers: {
        //预设好的header信息
        HttpHeaders.acceptHeader: "*/*",
        HttpHeaders.contentTypeHeader: "multipart/form-data;charset=UTF-8"
      },
      responseType: ResponseType.json,
    );

    //创建dio实例
    _dio = new Dio(_baseOptions);
    _dio.httpClientAdapter = Http2Adapter(
      ConnectionManager(
        idleTimeout: 10000,
        // Ignore bad certificate
        onClientCreate: (_, config) => config.onBadCertificate = (_) => true,
      ),
    );

    //开启请求和返回日志
//    _dio.interceptors.add(LogInterceptor());
    //可根据项目需要选择性的添加请求拦截器
//    _dio.interceptors.add(
//      InterceptorsWrapper(onRequest: (RequestOptions requestOptions) async {
//        //此处可网络请求之前做相关配置，比如会所有请求添加token，或者userId
//        requestOptions.queryParameters["token"] = "testtoken123443423";
//        requestOptions.queryParameters["userId"] = "123456";
//        print('-----请求参数--'+requestOptions.queryParameters.toString());
//        return requestOptions;
//      }, onResponse: (Response response) {
//        //此处拦截工作在数据返回之后，可在此对dio请求的数据做二次封装或者转实体类等相关操作
//        return response;
//      }, onError: (DioError error) {
//        //处理错误请求
//        return error;
//      }),
//    );
  }
后台接口地址是完整的：/member/checklist
通过postman请求：https://192.168.128.120:8443/app/member/checklist 可以正常返回列表
但是通过Dio请求后台提示错误：
2020-08-12 17:19:09 ERROR FEBS-Gateway boundedElastic-5 org.springframework.boot.autoconfigure.web.reactive.error.AbstractErrorWebExceptionHandler [5e9a238d-5]  500 Server Error for HTTP PRI ""
org.springframework.web.server.ResponseStatusException: 404 NOT_FOUND
	at org.springframework.web.reactive.resource.ResourceWebHandler.lambda$handle$0(ResourceWebHandler.java:325)
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Error has been observed at the following site(s):
	|_ checkpoint ⇢ org.springframework.cloud.gateway.filter.WeightCalculatorWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.security.web.server.authorization.AuthorizationWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.security.web.server.authorization.ExceptionTranslationWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.security.web.server.authentication.logout.LogoutWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.security.web.server.savedrequest.ServerRequestCacheWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.security.web.server.context.SecurityContextServerWebExchangeWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.security.web.server.context.ReactorContextWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.security.web.server.header.HttpHeaderWriterWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.security.config.web.server.ServerHttpSecurity$ServerWebExchangeReactorContextWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.security.web.server.WebFilterChainProxy [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.boot.actuate.metrics.web.reactive.server.MetricsWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ cc.mrbird.febs.gateway.enhance.auth.WebFluxSecurityCorsFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ HTTP null "" [ExceptionHandlingWebHandler]
大概步骤都是可以了，再换成https之前所有请求都没有问题。。。。感觉还是https的问题，可无从下爪啊。。。
Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by liuhengj** [comment]
这个问题前几天遇到过，我的情况是postman请求成功的地址和dio实际请求的地址进行对比，发现dio里面的某个字符串后面多个空格，而这个空格肉眼分辨不出来，你可以用vscode或者其他程序对比，滑动选中的时候两个字符串不相等，如果还有疑问可以发到197564178@qq.com
